lexicon/providers/aliyun.py

```diff
@@ -0,0 +1,217 @@
+"""Module provider for Aliyun"""
+from __future__ import absolute_import
+import logging
+import time
+import datetime
+import random
+import hmac
+import sys
+from hashlib import sha1
+from six.moves import urllib
+
+import requests
+from lexicon.providers.base import Provider as BaseProvider
+
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ['hichina.com']
+
+ALIYUN_DNS_API_ENDPOINT = 'https://alidns.aliyuncs.com'
+ALIYUN_DNS_DEFAULT_REGION = 'cn-hangzhou'
+
+def provider_parser(subparser):
```

Comment:
A good practice is also to fill the `description` of `subparser`. It allows to have a general description of the provider that is displayed when `lexicon provider --help` is invoked. It is a good place, for instance, to have a general description of the authentication used by the provider, and so what are the necessary sensible data that needs to be gathered and how. See for instance the provider `ovh` or `zeit`.

This is not mandatory at all, so if you consider that it does not worth it, I will not block on it.